microsoft/testsuites/kdump/kdumpcrash.py

```diff
@@ -223,6 +242,133 @@ def verify_kdumpcrash_large_memory_auto_size(
         self.is_auto = True
         self._kdump_test(node, log_path, log)
 
+    @TestCaseMetadata(
+        description="""
+        This test case will
+        1. replace FRE bins with CHK bins and reboot the VM
+        2. Configure kdump and reboot VM
+        3. Generate Crash with hvdbg syscall and verify dump
+
+        The test expects the directory containing MSHV CHK binaries tar to be passed
+        in the mshv_chk_bin / mshv_chk_loader testcase variables.
+        """,
+        priority=2,
+    )
+    def verify_mshv_crash(
+        self,
+        log: Logger,
+        node: Node,
+        variables: Dict[str, Any],
+        log_path: Path,
+    ) -> None:
+        chkbinpath = variables.get("mshv_chk_bin", "")
+        chkloaderpath = variables.get("mshv_chk_loader", "")
+        if not chkbinpath or not chkloaderpath:
+            raise SkippedException(
+                "Requires a path to MSHV binaries to be passed. "
+                "Please set mshv_chk_bin and mshv_chk_loader testcase variable."
+            )
+        if not isinstance(node.os, CBLMariner):
+            raise SkippedException(
+                f"Testcase only support Mariner-MSHV. Found: {node.os}"
+            )
+
+        mshv = node.tools[Ls].path_exists("/dev/mshv", sudo=True)
```

Comment:
If all above conditions meet, including the paths specified and it's the Mariner, can the test case just fail without mshv?

Reply:
yes, it is must to present. If /dev/mshv is not present means it is not running with our kernel. In that case, we will not have the sysfs entry present. Only Mariner-MSHV distro of ours will have it.

Reply:
Sorry, I didn’t express my point clearly earlier. The current logic skips the test case if /dev/mshv is not found. However, since the conditions above assume that MSHV-related files should exist, shouldn't the absence of /dev/mshv cause the test case to fail instead?

Also, the error message is a bit confusing. The node.os value is always "CBLMariner". It never shows as "Mariner-MSHV". Could we clarify that as well?

Reply:
This test intends to run with our private CBLMariner distro which comes with /dev/mshv. For other distro, this test should skip.

If test run with public CBLMariner distro, which do not have /dev/mshv, test should skip as well. Private mariner build of ours only will have the /dev/mshv available as it depends on the kernel we ship with private build.

I modified the exception msg now.

Reply:
I'm concerned that some bugs might be missed. For example, what happens if a build includes the MSHV kernel but omits this specific file? Is there a way to distinguish between an MSHV build that missed the file and a non-MSHV build?

My thinking is: if you have two separate pipelines for the different build types, or if you can pass in a variable indicating the build type, then you could always raise an exception here. But if there's no reliable way to differentiate the two build types and this step is skipped, then a missing file in an MSHV build might go undetected.

Reply:
When our kernel is up, it will always create /dev/mshv. If it is not present with our distro, that means something is wrong in kernel. We can check kernel logs in such cases. 

Only difference in public CBLMariner and private build is /dev/mshv. I can add some check of dmesg to verify if this is our private build or not, but it is just that, those messages would appear on successful creation of /dev/mshv only so we are checking the path. Similar piece of code is used for cloud-hypervisor perf test of LISA as well.

My thought process was, if we fail the test if /dev/mshv is not present then it will fail on public CBLMariner too which should ideally be skipped. I dont want to fail the test until necessary.